climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1165 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Define WildFire class.
+"""
+
+__all__ = ['WildFire']
+
+import logging
+import warnings
+import itertools
+from datetime import date
+from functools import partial
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+from scipy import sparse
+from sklearn.neighbors import BallTree
+from sklearn.cluster import DBSCAN
+import numba
+import geopandas as gpd
+from shapely.geometry import Point
+from shapely.ops import transform
+import pyproj
+
+from climada.hazard.centroids.centr import Centroids
+from climada.hazard.base import Hazard
+from climada.hazard.tag import Tag as TagHazard
+from climada.util.constants import ONE_LAT_KM, DEF_CRS
+from climada.util.dates_times import str_to_date
+
+from climada.util.alpha_shape import alpha_shape, plot_polygon
+warnings.simplefilter(action='ignore', category=FutureWarning)
+
+LOGGER = logging.getLogger(__name__)
+
+HAZ_TYPE = 'WF'
+""" Hazard type acronym for Wild Fire, might be changed to WFseason or WFsingle """
+
+CLEAN_THRESH = 30
+""" Minimal confidence value for the data from MODIS instrument to be use as input"""
+
+RES_DATA = 1.0
+""" Resolution of the data if no data origin provided (km) """
+
+BLURR_STEPS = 4
+""" steps with exponential decay for fire propagation matrix """
```

Comment:
Are these constant that should not be changed by a user, or are these constants that might change depending on the use case? If it is the latter, please remove them here and define them in the configuration file.

Reply:
I never really changed these constants, and they also don't really depend on a use case

Reply:
But these seem like something that people would like to have access too, for instance the BLURR_STEPS looks rather like a function parameter rather than a constant right?

Reply:
to be fair this part (the whole probabilistic part) will probably be altered in roughly half a year (with my master student coming in). Thus, I thought I'd put in a simple default in order for this to work

Reply:
That is a very dangerous thing to do. PLease make it clear what this default means and if it is something that should be tunable by the user, put it in the config file please.

Reply:
sure, I can add a warning. Is the correct config file for such variables `climada/conf/climada.conf`? and in the WF module I'd then load it by something like i.e. 
CLEAN_THRESH = CONFIG.hazard.wildfire.CLEAN_THRESH?

Reply:
in line with the comment below, I've included these in DataClasses - eaaf3b61

Reply:
Great Solution!